spec.emu

```diff
@@ -294,7 +294,8 @@ contributors: Google, Ecma International
             1. Add new part record { [[Type]]: `"literal"`, [[Value]]: _literal_ } as a new element of the list _result_.
           1. Let _p_ be the substring of _pattern_ from position _beginIndex_, exclusive, to position _endIndex_, exclusive.
           1. If _p_ matches a Property column of the row in <emu-xref href="#table-datetimeformat-components"></emu-xref>, then
-            1. Let _f_ be the value of _dateTimeFormat_'s internal slot whose name is the Internal Slot column of the matching row.
+            1. <ins>If _rangeFormatOptions_ is not *undefined*, let _f_ be the value of _rangeFormatOptions_'s field whose name is the Internal Slot column of the matching row.</ins>
```

Comment:
(Why doesn't GH allow me to annotate a line which wasn't modified in this PR? Sigh.)

Later one there's a step containing:
> If p is "month", then the String value may also depend on whether dateTimeFormat has a [[Day]] internal slot. If p is "timeZoneName", then the String value may also depend on the value of the [[inDST]] field of tm. If p is "era", then the String value may also depend on whether dateTimeFormat has a [[Era]] internal slot and if the implementation does not have a localized representation of f, then use f itself.

Let's ignore that "has a [[Day]] internal slot" is a bug in ECMA-402 (internal slots are always present; this should instead check that the internal slot's value isn't `undefined`). That part probably now needs to include the case when `rangeFormatOptions` is used instead of `dateTimeFormat`.

(Hmm, I don't understand the "era" handling in that step. Specifically, "if the implementation does not have a localized representation of f, then use f itself", but `f` is one of "narrow", "short", or "long". And I'm pretty sure these strings shouldn't appear verbatim in the output...)

Reply:
Thanks for pointing this out! I had missed this. I updated this paragraph to account for _rangeFormatOptions_, PTAL.

> Let's ignore that "has a [[Day]] internal slot" is a bug in ECMA-402 (internal slots are always present; this should instead check that the internal slot's value isn't undefined). 

I updated this instance of this error.

> (Hmm, I don't understand the "era" handling in that step. Specifically, "if the implementation does not have a localized representation of f, then use f itself", but f is one of "narrow", "short", or "long". And I'm pretty sure these strings shouldn't appear verbatim in the output...)

This sounds like a bug in the the spec, you are right that the value of f is not suitable (and doesn't make sense) for the output. I'll file a bug for this.